### ISSUE TITLE
Use RHEL9/CentOS stream dependencies for content resolver

### ISSUE THREAD

**Issue by voxik** [minimization/content-resolver-input#490]
It would be nice if content resolver was smart enough to use RHEL9/CentOS stream to resolve its dependencies instead of placeholders. E.g. for git-lfs, there is following placeholder:

https://github.com/minimization/content-resolver-input/blob/253dabe7f9a1f750be471c585d008e4498a778c4/configs/sst_cs_apps-scm.yaml#L18-L28

But it would be much easier, if content resolver knew where to ask RHEL9/CentOS stream for the real values from the package, which is not inherited into ELN.

**Comment by voxik** [comment]
Maybe it would be good start if there was tool, which could based on SRPM/RPMS provide content for the placeholder section, so this would be less manual (error prone) process.

**Comment by asamalik** [comment]
That's already possible, you just need to split the workload into two.

The one for ELN will only use the "eln" label, and the other one only "c9s".

**Comment by voxik** [comment]
> That's already possible, you just need to split the workload into two.

Oh, could you be please more specific? How we could do that with the git-lfs for example?

**Comment by asamalik** [comment]
Copy the existing one to a second file, perhaps `sst_cs_apps-scm-eln.yaml` and only leave the "eln" label (remove the "c9s" one). That will now be just ELN.

In the other one (`sst_cs_apps-scm.yaml`) remove the "eln" label and only keep the "c9s". And move all placeholders into "packages":


```
...
packages:
- git-lfs
...
```

**Comment by asamalik** [comment]
I split these two for example:

[sst_cs_infra_services-varnish-eln.yaml](https://github.com/minimization/content-resolver-input/blob/master/configs/sst_cs_infra_services-varnish-eln.yaml)

[sst_cs_infra_services-varnish.yaml](https://github.com/minimization/content-resolver-input/blob/master/configs/sst_cs_infra_services-varnish.yaml)

**Comment by voxik** [comment]
Thx for explanation. @opohorel could you please try to adjust the git-lfs? Not sure we have it in CentOS streams though, something to explore.

**Comment by opohorel** [comment]
I made a PR with proposed change. Could you please review it and merge if everything is ok. Thank you

**Comment by voxik** [comment]
Where I can see the impact of the PR above? Because I am afraid I don't understand what it actually does. I thought that we will be able to completely get rid of the placeholder ....

**Comment by voxik** [comment]
Now I probably realized how this is supposed to work. The c9s will be created based on CentOS stream as soon as it is available. Where ELN is actually RHEL10.

Anyway, I'd still prefer if the content resolver was more flexible and allowed to use the some "real" package dependencies instead of completely manual specification of them.

**Comment by asamalik** [comment]
Yes, exactly.

You can either use a real package, or this placeholder that you need to define manually. Are you hoping for something in between? What would that be?

**Comment by voxik** [comment]
Yes, as I said, I'd hope that there can be exception in ELN content set, which would take the dependencies from C9S. Or there would be at least tool to (re)generate the placeholder from (S)RPM.

**Comment by voxik** [comment]
Yet another idea, would it be possible to filter the dependencies?

**Comment by yselkowitz** [comment]
As stated above, if you want a placeholder in ELN to be handled as a regular package in CS, then you just need to split the configs.

However, after the golang vendoring-by-default change, git-lfs is no longer a placeholder in ELN.  Therefore, closing.
